homeassistant/components/gardena_bluetooth/__init__.py

```diff
@@ -0,0 +1,90 @@
+"""The Gardena Bluetooth integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from bleak.backends.device import BLEDevice
+from gardena_bluetooth.client import CachedConnection, Client
+from gardena_bluetooth.const import DeviceConfiguration, DeviceInformation
+from gardena_bluetooth.exceptions import CommunicationFailure
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import DeviceInfo
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import Coordinator
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+LOGGER = logging.getLogger(__name__)
+TIMEOUT = 20.0
+DISCONNECT_DELAY = 5
+
+
+class DeviceUnavailable(CommunicationFailure):
```

Comment:
This one seems like its blurring the lines of separating the responsibilities of the lib and HA by subclassing an exception from the lib